sled-storage/src/manager.rs

```diff
@@ -315,191 +376,363 @@ impl StorageManager {
-                    "disk_id" => ?raw_disk.identity()
+
+            info!(self.log, "No configuration exists");
+
+            // NOTE: What follows is an exceptional case: one where we have
+            // no record of "Control Plane Physical Disks", but we have zpools
+            // on our U.2s, and we want to use them regardless.
+            //
+            // THIS WOULD NORMALLY BE INCORRECT BEHAVIOR. In the future, these
+            // zpools will not be "automatically imported", and instead, we'll
+            // let Nexus decide whether or not to reformat the disks.
+            //
+            // However, because we are transitioning from "the set of disks /
+            // zpools is implicit" to a world where that set is explicit, this
+            // is a necessary transitional tool.
+            //
+            // TODO: Once we are confident that we have migrated to a world
+            // where this ledger is universally used, we should remove the
+            // following kludge.
+
+            let mut synthetic_config = vec![];
+            for (identity, disk) in self.resources.disks().values.iter() {
+                match disk {
+                    crate::resources::ManagedDisk::Unmanaged(raw) => {
+                        let zpool_path = match raw.u2_zpool_path() {
+                            Ok(zpool_path) => zpool_path,
+                            Err(err) => {
+                                info!(self.log, "Cannot find zpool path"; "identity" => ?identity, "err" => ?err);
+                                continue;
+                            }
+                        };
+
+                        let zpool_name =
+                            match sled_hardware::disk::check_if_zpool_exists(
+                                &zpool_path,
+                            ) {
+                                Ok(zpool_name) => zpool_name,
+                                Err(err) => {
+                                    info!(self.log, "Zpool does not exist"; "identity" => ?identity, "err" => ?err);
+                                    continue;
+                                }
+                            };
+
+                        info!(self.log, "Found existing zpool on device without ledger";
+                            "identity" => ?identity,
+                            "zpool" => ?zpool_name);
+
+                        // We found an unmanaged disk with a zpool, even though
+                        // we have no prior record of a ledger of control-plane
+                        // disks.
+                        synthetic_config.push(
+                            // These disks don't have a control-plane UUID --
+                            // report "nil" until they're overwritten with real
+                            // values.
+                            OmicronPhysicalDiskConfig {
+                                identity: identity.clone(),
+                                id: Uuid::nil(),
```

Comment:
This is a bit hacky, but I don't really know of a better option here. There is no sled-local "control plane UUID" of the physical disk to be used on existing systems, so rather than lie, I am using "nil".